client/fvESg_service.go

```diff
@@ -0,0 +1,389 @@
+	dn := fmt.Sprintf(models.Dn, tenant ,application_profile ,name )    
+	cont, err := sm.Get(dn)
+	if err != nil {
+		return nil, err
+	}
+	fvESg := models.EndpointSecurityGroupFromContainer(cont)
+	return fvESg, nil
+}
+
+func (sm *ServiceManager) DeleteEndpointSecurityGroup(name string ,application_profile string ,tenant string ) error {
+	dn := fmt.Sprintf(models.Dn, tenant ,application_profile ,name )
+	return sm.DeleteByDn(dn, models.FvesgClassName)
+}
+
+func (sm *ServiceManager) UpdateEndpointSecurityGroup(name string ,application_profile string ,tenant string  ,description string, fvESgattr models.EndpointSecurityGroupAttributes) (*models.EndpointSecurityGroup, error) {
+	rn := fmt.Sprintf(models.Rn,name)
+	parentDn := fmt.Sprintf(models.ParentDn, tenant ,application_profile )
+	fvESg := models.NewEndpointSecurityGroup(rn, parentDn, description, fvESgattr)
+    fvESg.Status = "modified"
+	err := sm.Save(fvESg)
+	return fvESg, err
+}
+
+func (sm *ServiceManager) ListEndpointSecurityGroup(application_profile string ,tenant string ) ([]*models.EndpointSecurityGroup, error) {
+	dnUrl := fmt.Sprintf("%s/models.ParentDn/fvESg.json", models.BaseurlStr , tenant ,application_profile )
+    cont, err := sm.GetViaURL(dnUrl)
+	list := models.EndpointSecurityGroupListFromContainer(cont)
+	return list, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsSecInheritedFromEndpointSecurityGroup( parentDn, tDn string) error {
+	dn := fmt.Sprintf("%s/rssecInherited-[%s]", parentDn, tDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"				
+			}
+		}
+	}`, "fvRsSecInherited", dn))
+	jsonPayload, err := container.ParseJSON(containerJSON)
+	if err != nil {
+		return err
+	}
+	req, err := sm.client.MakeRestRequest("POST", fmt.Sprintf("%s.json", sm.MOURL), jsonPayload, true)
+	if err != nil {
+		return err
+	}
+	cont, _, err := sm.client.Do(req)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%+v", cont)
+	return nil
+}
+
+func (sm *ServiceManager) DeleteRelationfvRsSecInheritedFromEndpointSecurityGroup(parentDn , tDn string) error{
+	dn := fmt.Sprintf("%s/rssecInherited-[%s]", parentDn, tDn)
+	return sm.DeleteByDn(dn , "fvRsSecInherited")
+}
+
+func (sm *ServiceManager) ReadRelationfvRsSecInheritedFromEndpointSecurityGroup( parentDn string) (interface{},error) {
+	dnUrl := fmt.Sprintf("%s/%s/%s.json",models.BaseurlStr,parentDn,"fvRsSecInherited")
+	cont, err := sm.GetViaURL(dnUrl)
+	contList := models.ListFromContainer(cont,"fvRsSecInherited")
+	st := &schema.Set{
+		F: schema.HashString,
+	}
+	for _, contItem := range contList{
+		dat := models.G(contItem, "tDn")
+		st.Add(dat)
+	}
+	return st, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsProvFromEndpointSecurityGroup( parentDn, tnVzBrCPName string) error {
+	dn := fmt.Sprintf("%s/rsprov-%s", parentDn, tnVzBrCPName)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"				
```

Comment:
We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.